tool/site/client-react/src/components/sorter.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {ReactNode} from "react"
+import {ChevronLeft, ChevronRight} from "nhsuk-react-components/lib/components/icons"
+import styled from "styled-components"
+
+export interface SortConfig<T> {
+  key: keyof T
+  ascending: boolean
+}
+
+export interface Sorter<T> {
+  sortedItems: Array<T>
+  sortBy: (key: keyof T) => void
+  getIcon: (key: keyof T) => ReactNode
+}
+
+export const useSorter = <T extends unknown>(items: Array<T>, config: SortConfig<T> = null): Sorter<T> => {
+  const [sortConfig, setSortConfig] = React.useState<SortConfig<T>>(config)
+
+  const sortedItems = React.useMemo(() => {
+    const itemsCopy = [...items]
+    if (sortConfig !== null) {
+      itemsCopy.sort((a, b) => {
+        const sortValueA = a[sortConfig.key]
+        const sortValueB = b[sortConfig.key]
+        const directionValue = sortConfig.ascending ? 1 : -1
+        if (sortValueA > sortValueB) {
+          return directionValue
+        } else if (sortValueA < sortValueB) {
+          return -directionValue
+        } else {
+          return 0
+        }
+      })
+    }
+    return itemsCopy
+  }, [items, sortConfig])
+
+  const sortBy = (key: keyof T) => {
+    if (sortConfig && sortConfig.key === key) {
+      setSortConfig({key, ascending: !sortConfig.ascending})
+    } else {
+      setSortConfig({key, ascending: true})
+    }
+  }
+
+  const getIcon = (key: keyof T) => {
+    if (sortConfig?.key !== key) {
+      return null
+    }
+    return sortConfig.ascending ? <ChevronUpMini/> : <ChevronDownMini/>
+  }
+
+  return {sortedItems, sortBy, getIcon}
+}
+
+const ChevronUpMini = styled(ChevronLeft)`
+  transform: rotate(90deg);
+  width: 1em;
```

Comment:
Don't like setting the size manually, but I'm not good enough at CSS to be able to say "make it the same size as the text please"

Reply:
At least using a size in em means the icon should scale with the font size